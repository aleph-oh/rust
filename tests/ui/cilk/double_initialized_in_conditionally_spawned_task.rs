#![feature(cilk)]
// Tests that when we sync conditionally spawned tasks, the liveness analysis correctly considers
// the variables re-written to. In conjunction with the other tests for conditionally spawned tasks,
// this should restrict implementations only to those which accurately track which tasks are synced
// at a sync.

// compile-flags: -C panic=abort
// no-prefer-dynamic

fn f() -> usize {
    let y = 5;
    if true {
        cilk_spawn { y = 1; };
    //~^ ERROR cannot assign twice to immutable variable `y` [E0384]
    }
    cilk_sync;
    y + 1
}

fn main() {
}
