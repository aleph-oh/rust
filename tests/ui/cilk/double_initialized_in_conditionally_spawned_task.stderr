error[E0384]: cannot assign twice to immutable variable `y`
  --> $DIR/double_initialized_in_conditionally_spawned_task.rs:13:22
   |
LL |     let y = 5;
   |         -
   |         |
   |         first assignment to `y`
   |         help: consider making this binding mutable: `mut y`
LL |     if true {
LL |         cilk_spawn { y = 1; };
   |                      ^^^^^ cannot assign twice to immutable variable

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0384`.
